drivers/i2s/i2s_mcux_sai.c

```diff
@@ -0,0 +1,1172 @@
+/*
+ * Copyright (c) 2021 NXP Semiconductor INC.
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/** @file
+ * @brief I2S bus (SAI) driver for NXP i.MX RT series.
+ */
+
+#include <errno.h>
+#include <string.h>
+#include <sys/__assert.h>
+#include <kernel.h>
+#include <device.h>
+#include <init.h>
+#include <drivers/dma.h>
+
+#include <drivers/i2s.h>
+#include <drivers/pinmux.h>
+#include <drivers/clock_control.h>
+#include <dt-bindings/clock/imx_ccm.h>
+#include <soc.h>
+
+#include "i2s_mcux_sai.h"
+
+#define LOG_DOMAIN dev_i2s_mcux
+#define LOG_LEVEL CONFIG_I2S_LOG_LEVEL
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(LOG_DOMAIN);
+
+#define DT_DRV_COMPAT nxp_mcux_i2s
+
+/*
+ * This indicates the Tx/Rx stream.
+ *
+ * in_queue and out_queue are used as follows
+ *   transmit stream:
+ *   application provided buffer is queued to in_queue until loaded to DMA.
+ *   when DMA channel is idle, buffer is retrieved from in_queue and loaded
+ *   to DMA and queued to out_queue. when DMA completes, buffer is retrieved
+ *   from out_queue and freed.
+ *
+ *   receive stream:
+ *   driver allocates buffer from slab and loads DMA buffer is queued to
+ *   in_queue when DMA completes, buffer is retrieved from in_queue
+ *   and queued to out_queue when application reads, buffer is read
+ *   (may optionally block) from out_queue and presented to application.
+ */
+struct stream {
+	int32_t state;
+	uint32_t dma_channel;
+	uint32_t start_channel;
+	void (*irq_call_back)(void);
+	struct i2s_config cfg;
+	struct dma_config dma_cfg;
+	struct dma_block_config dma_block;
+	struct k_msgq in_queue;
+	void *in_msgs[CONFIG_I2S_RX_BLOCK_COUNT];
+	struct k_msgq out_queue;
+	void *out_msgs[CONFIG_I2S_TX_BLOCK_COUNT];
+	uint32_t stream_starving;
+};
+
+struct i2s_mcux_config {
+	I2S_Type *base;
+	uint32_t bus_id;
```

Comment:
Thanks for updating this driver so it doesn't use hard-coded DT_DRV_INST numbers anymore.

This `bus_id` value used to be used in the `switch` statement that contained the DT_DRV_INST usage before. I copy/pasted some of the old code here: https://github.com/zephyrproject-rtos/zephyr/pull/37335#issuecomment-964817122

Now in the update the `switch` is gone. But I don't see `bus_id` used anywhere.

Is it OK?

Reply:
yes, the bus_id can be removed. let me update, thanks for pointing this out.